convert2rhel/actions/post_conversion/hostmetering.py

```diff
@@ -0,0 +1,238 @@
+            self.set_result(
+                level="ERROR",
+                id="HOST_METERING_NOT_RUNNING",
+                title="Host metering service is not running.",
+                description="host-metering.service is not running.",
+                remediations="You can try to start the service manually"
+                " by running following command:\n"
+                "   systemctl start host-metering.service",
+            )
+
+        return service_running
+
+    def _check_env_var(self):
+        """Check if the env var is set and if it has the right content. If the
+        content is auto|force, the hostmetering should be configued on the system.
+
+        :return: Return True if the value is equal to auto|force, otherwise False
+        :rtype: bool
+        """
+        if self.env_var is None:
+            logger.debug("CONVERT2RHEL_CONFIGURE_HOST_METERING was not set. Skipping it.")
+            self.add_message(
+                level="INFO",
+                id="CONFIGURE_HOST_METERING_SKIP",
+                title="Did not perform host metering configuration.",
+                description="CONVERT2RHEL_CONFIGURE_HOST_METERING was not set.",
+            )
+            return False
+
+        if self.env_var not in ("force", "auto"):
+            logger.debug("Value for environment variable not recognized: %s" % self.env_var)
+            self.add_message(
+                level="WARNING",
+                id="UNRECOGNIZED_OPTION_CONFIGURE_HOST_METERING",
+                title="Unrecognized option in CONVERT2RHEL_CONFIGURE_HOST_METERING environment variable.",
+                description="Environment variable {env_var} not recognized.".format(env_var=self.env_var),
+                remediations="Set the option to `auto` value if you want to configure host metering.",
+            )
+            return False
+
+        if self.env_var == "force":
+            logger.warning(
+                "The `force' option has been used for the CONVERT2RHEL_CONFIGURE_HOST_METERING environment variable."
+                " Please note that this option is mainly used for testing and will configure host-metering unconditionally. "
+                " For generic usage please use the 'auto' option."
+            )
+            self.add_message(
+                level="WARNING",
+                id="FORCED_CONFIGURE_HOST_METERING",
+                title="The `force' option has been used for the CONVERT2RHEL_CONFIGURE_HOST_METERING environment variable.",
+                description="Please note that this option is mainly used for testing and"
+                " will configure host-metering unconditionally."
+                " For generic usage please use the 'auto' option.",
+            )
+        elif self.env_var == "auto":
+            logger.debug("Automatic detection of host hyperscaler and configuration.")
+
+        return True
+
+    def is_running_on_hyperscaler(self, rhsm_facts):
```

Comment:
This can be moved to the module level instead of the class level since it does not depend on anything from the class (self). 

We can make it private as well to indicate that this is intended to be used only inside this module